habushu_cucumber_formatter/__init__.py

```diff
@@ -0,0 +1,270 @@
+        step['embeddings'].append({
+            'mime_type': mime_type,
+            'data': base64.b64encode(data).replace('\n', ''),
+        })
+ 
+    def eof(self):
+        """
+        End of feature
+        """
+        if not self.current_feature_data:
+            return
+ 
+        # -- NORMAL CASE: Write collected data of current feature.
+        self.update_status_data()
+ 
+        if self.feature_count == 0:
+            # -- FIRST FEATURE:
+            self.write_json_header()
+        else:
+            # -- NEXT FEATURE:
+            self.write_json_feature_separator()
+ 
+        self.write_json_feature(self.current_feature_data)
+        self.current_feature_data = None
+        self.feature_count += 1
+ 
+    def close(self):
+        self.write_json_footer()
+        self.close_stream()
+ 
+    # -- JSON-DATA COLLECTION:
+    def add_feature_element(self, element):
+        assert self.current_feature_data is not None
+        if 'elements' not in self.current_feature_data:
+            self.current_feature_data['elements'] = []
+        self.current_feature_data['elements'].append(element)
+        return element
+ 
+    @property
+    def current_feature_element(self):
+        assert self.current_feature_data is not None
+        return self.current_feature_data['elements'][-1]
+ 
+    @property
+    def current_step(self):
+        step_index = self._step_index
+        if self.current_feature.background is not None:
+            element = self.current_feature_data['elements'][-2]
+            if step_index >= len(self.current_feature.background.steps):
+                step_index -= len(self.current_feature.background.steps)
+                element = self.current_feature_element
+        else:
+            element = self.current_feature_element
+ 
+        return element['steps'][step_index]
+ 
+    def update_status_data(self):
+        assert self.current_feature
+        assert self.current_feature_data
+        self.current_feature_data['status'] = self.current_feature.status.name
+ 
+    def write_tags(self, tags):
+        return [{'name': f'@{tag}', 'line': tag.line if hasattr(tag, 'line') else 1} for tag in tags]
+ 
+    def generate_id(self, feature, scenario=None):
+        def convert(name):
+            return name.lower().replace(' ', '-')
+        id = convert(feature.name)
+        if scenario is not None:
+            id += ';'
+            id += convert(scenario.name)
+        return id
+ 
+    def format_description(self, lines):
+        description = '\n'.join(lines)
+        description = '<pre>%s</pre>' % description
+        return description
+ 
+    # -- JSON-WRITER:
+    def write_json_header(self):
+        self.stream.write('[\n')
+        self.hasWrittenHeader = True
+ 
+    def write_json_footer(self):
+        if self.hasWrittenHeader:
+            self.stream.write('\n]\n')
+        else:
+            self.stream.write('')
+        #     self.stream.write('''
```

Comment:
Will remove the commented section (rename is being interpreted as a new file). Main change is the addition of the hasWrittenHeader class member and the check for it in `write_json_footer` to make sure file content stays empty if not header was written.